### ISSUE TITLE
[Java/Jersey-1.18] Support additional property to declare a list of used dependencies in generation configuration file.

### ISSUE THREAD

**Issue by wuchenwang** [swagger-api/swagger-codegen#3196]
##### Description

We are using the swagger-codegen jersey-1.18 to generate a server stub that different service team could use to hook their own implementation to, in doing that, they will need to declare dependencies in their build assembly file. Currently swagger-codegen does not support any customizable dependency declaration in `pom.xml`, we would like to have this feature where we could use the generation configuration file to add these dependencies.

example:
``` json
{
    "lang" : "java",
    "inputSpec" : "swagger.yaml",
    "outputDir" : "src/gen/java",
    "verbose" : true,
    "skipOverwrite" : true,
    "templateDir" : "src/main/resources",
    "auth" : "hello:world",
    "apiPackage" : "io.something.api",
    "modelPackage" : "io.something.models",
    "invokerPackage" : "io.something.invoker",
    "groupId" : "io.something",
    "artifactId" : "awesome-api",
    "artifactVersion" : "1.2.3",
    "library" : "jersey2",
    "languageSpecificPrimitives" : [ "rolex" ],
    "localVariablePrefix" : "_",
    "dependencies" : [
        {"groupId" : "A",
        "artifactId" : "A1"},
        {"groupId" : "B",
        "artifactId" : "B1"}
    ]
}
```

##### Swagger-codegen version

2.2-SNAPSHOT

**Comment by wing328** [comment]
@wuchenwang one way is to use vendor extension: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#vendorExtensions and then use customized template with `-t` option.

Would that meet your requirement?

**Comment by wuchenwang** [comment]
@wing328 If I understand correctly, this vendorExtension seems to add attributes to schema of a swagger api spec. I think using api to generate a service stub is sort of an implementation detail (because you can choose not to generate and implement the api-spec in your own way), so I would rather changes such as declaring dependencies the implementation is going to use be scoped to the api spec implementation, in this case `swagger-codegen`.

We are currently already using `-t` to make use of our own template, but still need those template variables injected properly.

**Comment by cbornet** [comment]
@wuchenwang This seems really specific to your use case and I don't think this is useful for everybody. In the hand there will be too much options used by too few people with a huge maintenance cost. Why don't you just use your own pom.xml that you use in place of the generated one ?

**Comment by wuchenwang** [comment]
@cbornet  Yes I could definitely use my own pom.xml, what I would like to do is making it customizable, in order to do so some configuration change with codegen is needed.

**Comment by cbornet** [comment]
Can you describe more in details what you are trying to do ? It's not clear to me why you would add dependencies to the pom that are not used in the generated code. If it's because you then add business logic that requires new dependencies, then just add them by editing the generated pom !

**Comment by wuchenwang** [comment]
Sorry for the confusion, I didn't make my use case clear in the description. We want new service developers to easily prototype their service by defining a contract (api-spec) first and could easily spin up a service by some kind of generation, in doing so we also want them to get out of the business of wiring up their own dependencies, pom definition etc. to focus on writing business logic, something like [AWS lambda](http://docs.aws.amazon.com/lambda/latest/dg/welcome.html) function. 

So the added dependencies could be declared before the generation (wiring up the service) upfront, prior to business logic is written in those lambda functions, and they do not add unnecessary dependencies on the fly, because they do not own the build (pom.xml).

I hope this clears some of the mud.

**Comment by fehguy** [comment]
@wuchenwang it sounds like what you really need is to have custom templates for exactly what your use case is looking for, so you can do what you're describing.  The codegen is a great starting place and has mechanisms for what you want to do, but what you describe doesn't sound like a normal use case that we're trying to solve with this project.

**Comment by cbornet** [comment]
It looks like you would need another tool to assemble the dependencies needed by swagger and the ones for your service. The pom is in xml so it's really not hard to do it by a script.
